### PR TITLE
Add pretrained ResNet50 on ImageNet

### PR DESCRIPTION
**Pull request by arrufat**
As discussed in #29 here's the pretrained ResNet50 on ImageNet.

I've also included the `std::vector<string> labels` as in the ResNet34 file.

Moreover, I've put a small example on how to use the pretrained backbone in another network and how to use the `resnet.h` definitions on a new network.

### CODE REVIEW

**Comment by arrufat** [comment]
I realized I made a mistake and uploaded an "affine" version instead, so I updated the model. Please, make sure to squash the merge, so that the repository size doesn't explode 😅

**Comment by arrufat** [comment]
I got a bit carried away on the usage example and added some extra information on how to use the backbone on different network architectures and how to change the learning rate of particular layers for fine-tuning... Maybe I should transform it into an example? something like `dnn_introduction3_ex.cpp`?

**Comment by davisking** [comment]
Yeah that sounds like good stuff for a 3rd intro example :)

**Comment by arrufat** [comment]
So maybe I should close this PR open a new one on the dlib repo.
Do you still want to put the file here? If that's the case, I will remove everything except for the trained model.

**Comment by davisking** [comment]
The trained model files should definitely go in https://github.com/davisking/dlib-models.  And if there is any code you want to include with them that isn't part of the `dnn_introduction3_ex.cpp` then it can go here.  Whatever you think is sensible.

**Comment by arrufat** [comment]
Good! I am thinking about doing the same thing as in the semantic segmentation examples, where the main code is in the `cpp` files, but the model is defined in a header file. And in this repository, only the resnet50 model :)

**Comment by davisking** [comment]
Thanks for the PR.  I also put a copy of the file at dlib.net/files
